Add check to only create native enum once

mogzol opened the pull request:
This will fix #3657

kibertoad commented:
Can you also add test for this? I think we had integration tests for enums already, should be easy to expand.

elhigu commented on lib/dialects/postgres/schema/columncompiler.js:
```diff
@@ -8,6 +8,7 @@ const { isObject } = require('lodash');
 function ColumnCompiler_PG() {
   ColumnCompiler.apply(this, arguments);
   this.modifiers = ['nullable', 'defaultTo', 'comment'];
+  this._nativeEnumCreated = false;
```
Are you sure this works, when there are multiple native enums declared / altered in the same table? This is a column compiler, so it probably does, but anyways set of tests for this would be needed. Also other tests than just having the most trivial happy path tested (cases where multiple columns are created and some altered)

mogzol replied:
Added some tests

kibertoad commented:
@mogzol Do you need any help with this?

mogzol commented:
Oh sorry I forgot about this actually. I don't really have much free time right now if someone else want's to take this over, or wait until closer to the end of the year and I should have a chance to fix it.

mogzol commented:
Updated with some tests and a slightly different approach, now the result of the type call in `getColumnType` in the column compiler is cached so it's only run once (per column), which should fix this issue for any types that have side effects, instead of just fixing the specific postgres enu type.

kibertoad commented:
@mogzol Could you also please add an integration test to `test/integration2`?

mogzol commented:
@kibertoad Yes, added some integration tests. I also found that altering native enum columns still failed because postgres doesn't allow casting directly between enum types, [they must be converted to text first](https://www.munderwood.ca/index.php/2015/05/28/altering-postgresql-columns-from-one-enum-to-another/), so I added a check in the postgres column compiler for that.

kibertoad commented:
Thank you!
